### ISSUE TITLE
Problem with installation on Ubuntu 18.04  - No such file or directory  #include "laszip_api.h"

### ISSUE THREAD

**Issue by ghost** [potree/PotreeConverter#327]
Hello guys, I am using Ubuntu 18.04 and I have the problem with installation. I followed the instruction which @spatialhast posted here: https://github.com/potree/PotreeConverter/issues/180 
I successfully installed LAStools, but I got stuck at the Portree. :/ When I run the command 

    sudo make && sudo make install

I get the output:

    Scanning dependencies of target PotreeConverter
     [  8%] Building CXX object 
    PotreeConverter/CMakeFiles/PotreeConverter.dir/src/BINPointReader.cpp.o
    [ 16%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/GridCell.cpp.o
    [ 25%] Building CXX object 
     PotreeConverter/CMakeFiles/PotreeConverter.dir/src/LASPointReader.cpp.o
  
  
      /home/user/LAStools/LASzip/build/PotreeConverter/PotreeConverter/src/LASPointReader.cpp:8:10: 
        fatal error: laszip_api.h: No such file or directory
         #include "laszip_api.h"
             ^~~~~~~~~~~~~~
         compilation terminated.
        PotreeConverter/CMakeFiles/PotreeConverter.dir/build.make:110: recipe for target 
        'PotreeConverter/CMakeFiles/PotreeConverter.dir/src/LASPointReader.cpp.o' failed
        make[2]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/src/LASPointReader.cpp.o] Error 1
        CMakeFiles/Makefile2:85: recipe for target 'PotreeConverter/CMakeFiles/PotreeConverter.dir/all' failed
        make[1]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/all] Error 2
        Makefile:129: recipe for target 'all' failed
        make: *** [all] Error 2

But I have that file in LAStools/LASzip/dll 
I tried to manually copy it in /LAStools/LASzip/build/PotreeConverter/build but without success.

**Comment by zxcvb54321** [comment]
The solution to my problem was, believe or not, just to run this command separately:

    sudo make && sudo make install 

So, I first run make, and then make install. 

Or maybe restart was the solution (But I am using Ubuntu, I don't know..). Because I tried to install again the day after and my computer was switched off in meantime. 

Anyway, these two things are only things that I changed from yesterday.

**Comment by swiss-knight** [comment]
You have to run cmake with these flags: `cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=/your_path/to/LAStools/LASzip/dll -DLASZIP_LIBRARY=/your_path/to/LAStools/LASzip/build/src/liblaszip.so ..`

**Comment by freshcoderchan** [comment]
When I run **make** command after running this command, I got an error.

`sudo cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=~/var/www/html/lastools/master/LASzip/dll -DLASZIP_LIBRARY=~/var/www/html/lastools/master/LASzip/build/src/liblaszip.so ..`

So I removed **~** in the path.

`sudo cmake -DCMAKE_BUILD_TYPE=Release -DLASZIP_INCLUDE_DIRS=/var/www/html/lastools/master/LASzip/dll -DLASZIP_LIBRARY=/var/www/html/lastools/master/LASzip/build/src/liblaszip.so ..`

It worked. 🔥

**Comment by erickucl** [comment]
Hello, I have the following error when executing the make on potreeconverter build:
sudo make
[  8%] Building CXX object PotreeConverter/CMakeFiles/PotreeConverter.dir/src/BINPointReader.cpp.o
In file included from /home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/include/BINPointReader.hpp:11,
                 from /home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:8:
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/include/PointReader.h:11:21: error: ‘filesystem’ is not a namespace-name
   11 | namespace fs = std::filesystem;
      |                     ^~~~~~~~~~
In file included from /home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:9:
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/include/stuff.h:38:21: error: ‘filesystem’ is not a namespace-name
   38 | namespace fs = std::filesystem;
      |                     ^~~~~~~~~~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/include/stuff.h:96:15: error: ‘fs’ has not been declared
   96 |  bool copyDir(fs::path source, fs::path destination);
      |               ^~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/include/stuff.h:96:32: error: ‘fs’ has not been declared
   96 |  bool copyDir(fs::path source, fs::path destination);
      |                                ^~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/include/stuff.h:96:52: error: expression list treated as compound expression in initializer [-fpermissive]
   96 |  bool copyDir(fs::path source, fs::path destination);
      |                                                    ^
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:11:21: error: ‘filesystem’ is not a namespace-name
   11 | namespace fs = std::filesystem;
      |                     ^~~~~~~~~~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp: In constructor ‘Potree::BINPointReader::BINPointReader(std::string, Potree::AABB, double, Potree::PointAttributes)’:
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:27:5: error: ‘fs’ has not been declared
   27 |  if(fs::is_directory(path)){
      |     ^~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:30:7: error: ‘fs’ has not been declared
   30 |   for(fs::directory_iterator it(path); it != fs::directory_iterator(); it++){
      |       ^~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:30:40: error: ‘it’ was not declared in this scope; did you mean ‘int’?
   30 |   for(fs::directory_iterator it(path); it != fs::directory_iterator(); it++){
      |                                        ^~
      |                                        int
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:30:46: error: ‘fs’ has not been declared
   30 |   for(fs::directory_iterator it(path); it != fs::directory_iterator(); it++){
      |                                              ^~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:31:4: error: ‘fs’ has not been declared
   31 |    fs::path filepath = it->path();
      |    ^~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:32:7: error: ‘fs’ has not been declared
   32 |    if(fs::is_regular_file(filepath)){
      |       ^~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:32:27: error: ‘filepath’ was not declared in this scope
   32 |    if(fs::is_regular_file(filepath)){
      |                           ^~~~~~~~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp: In member function ‘virtual bool Potree::BINPointReader::readNextPoint()’:
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:182:5: error: ‘memcpy’ was not declared in this scope
  182 |     memcpy(target.data(), (buffer + offset), attribute.byteSize);
      |     ^~~~~~
/home/ubuntu/Desktop/workspaces/PotreeConverter/master/PotreeConverter/src/BINPointReader.cpp:10:1: note: ‘memcpy’ is defined in header ‘<cstring>’; did you forget to ‘#include <cstring>’?
    9 | #include "stuff.h"
  +++ |+#include <cstring>
   10 | 
make[2]: *** [PotreeConverter/CMakeFiles/PotreeConverter.dir/build.make:63: PotreeConverter/CMakeFiles/PotreeConverter.dir/src/BINPointReader.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:91: PotreeConverter/CMakeFiles/PotreeConverter.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

any hint of what could it be? regards

**Comment by il-Brando** [comment]
@erickucl 
I solved in ubuntu 19.10 the issue about ` namespace fs = std::filesystem;`
in all files where is used replacing following code 

```
#include <filesystem>

namespace fs = std::filesystem;
```

with this
```
//#include <filesystem>
#include <experimental/filesystem>

//namespace fs = std::filesystem;
namespace fs = std::experimental::filesystem;
```
and issue about  memcpy definition adding follow declaration at top of PotreeConverter/src/BINPointReader.cpp
```
#include <cstring>
```

**Comment by randomluck17** [comment]
@il-Brando Can you specify the files? I can't really find the file, especially BINPointReader.cpp since that's the one breaking for me.
